src/v/raft/consensus.cc

```diff
@@ -948,9 +956,7 @@ ss::future<vote_reply> consensus::do_vote(vote_request&& r) {
 ss::future<append_entries_reply>
 consensus::append_entries(append_entries_request&& r) {
     return with_gate(_bg, [this, r = std::move(r)]() mutable {
-        return _op_lock.with([this, r = std::move(r)]() mutable {
-            return do_append_entries(std::move(r));
-        });
+        return _append_requests_buffer.enqueue(std::move(r));
```

Comment:
i assume that this buffer has a semaphore of say 256 futures or smth like that ? we still need to propagate backpressure.